### PR TITLE
Fix #17399: Guest not having being watched thoughts

### LINKED ISSUE #17399: Guest thought "I have the strangest feeling someone is watching me" not appearing in OpenRCT2
### Operating System

Windows 10, 64 bit

### OpenRCT2 build

OpenRCT2, v0.4.0 release

### Describe the issue

When I was playing a bit of vanilla RCT2 and followed around some guests, they think a thought about being followed: "I have the strangest feeling someone is watching me". In the old days that was such a nice thing to see as a child.

However, in OpenRCT2 I could not remember seeing this thought at all; while in vanilla it shows up pretty quickly.
So I think this is a bug in OpenRCT2.

### Area(s) with issue?

_No response_

### Steps to reproduce

1. Open a park
2. Click on a guest to follow him around
3. Within a short time the thought "I have the strangest feeling someone is watching me" should appear, but it doesn't.

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by jazzysoggy**
Fix #17399 

Issue: Guests never have the thought of being watched, even though vanilla behavior has this happen after a while.

Fix: Decoupled w->highlighted_item from w->var_496, as w->var_496 is linked to the newAnimationFrame cycle. This would inadvertently cause the highlighted_item value to only reach a maximum of 24, as this cycle resets at 24, which is lower than the threshold required to even get past the first check to insert the thought.

### CODE REVIEW

File: src/openrct2-ui/windows/Guest.cpp [hunk 1]
```diff
@@ -32,6 +32,7 @@
 #include <openrct2/windows/Intent.h>
 #include <openrct2/world/Footpath.h>
 #include <openrct2/world/Park.h>
+#include <random>
```
Comment (rctdude2) [inline src/openrct2-ui/windows/Guest.cpp:35]: Forgot to remove this.

**Comment by rctdude2** [comment]
Simpler fix then mine (#17218). As far as I can tell, `highlighted_item` isn't used for anything in the guest window other than the watch timer, so this should work.

This probably needs a changelog, it's been broken for a while (refer to https://github.com/OpenRCT2/OpenRCT2/pull/17218#issuecomment-1127092752).

**Comment by Gymnasiast** [comment]
Superseded by https://github.com/OpenRCT2/OpenRCT2/pull/17471.

**Comment by duncanspumpkin** [comment]
has been superseded #17471.

**Pull request closed without merging**